timelock-api/src/main/java/com/palantir/atlasdb/timelock/api/GenericNamedMinTimestamp.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.api;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import org.immutables.value.Value;
+
+@JsonDeserialize(as = ImmutableGenericNamedMinTimestamp.class)
+@JsonSerialize(as = ImmutableGenericNamedMinTimestamp.class)
+@Value.Immutable
+public interface GenericNamedMinTimestamp {
+    String RESERVED_NAMED_MIN_TIMESTAMP_NAME_FOR_IMMUTABLE_TIMESTAMP = "ImmutableTimestamp";
+
+    @JsonValue
+    String name();
+
+    @Value.Check
+    default void check() {
+        Preconditions.checkArgument(
+                !name().equals(RESERVED_NAMED_MIN_TIMESTAMP_NAME_FOR_IMMUTABLE_TIMESTAMP),
+                "Name must not be a reserved name",
+                SafeArg.of("name", name()));
+    }
+
+    static GenericNamedMinTimestamp of(String name) {
+        return ImmutableGenericNamedMinTimestamp.builder().name(name).build();
+    }
+}
```

Comment:
This is from the server API PRs. I just put it here as it's better than having the PRs depend on one another